modules/contact/src/ContactSlipDamper.C

```diff
@@ -0,0 +1,226 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ContactSlipDamper.h"
+#include "FEProblem.h"
+#include "DisplacedProblem.h"
+#include "AuxiliarySystem.h"
+#include "PenetrationLocator.h"
+#include "NearestNodeLocator.h"
+
+template<>
+InputParameters validParams<ContactSlipDamper>()
+{
+  InputParameters params = validParams<GeneralDamper>();
+  params.addRequiredParam<std::vector<int> >("master", "IDs of the master surfaces for which slip reversals should be damped");
+  params.addRequiredParam<std::vector<int> >("slave", "IDs of the slave surfaces for which slip reversals should be damped");
+  params.addParam<Real>("max_iterative_slip", std::numeric_limits<Real>::max(), "Maximum iterative slip");
+  params.addParam<Real>("min_damping_factor", 0.0, "Minimum permissible value for damping factor");
```

Comment:
Might use `checkedRangeParam` just to ward off another potential user bug. There may be a few easy opportunities here for checkedRanges.